src/core/modules/impl/DefaultStateEvaluator.ts

```diff
@@ -287,9 +287,13 @@ export abstract class DefaultStateEvaluator implements StateEvaluator {
         }
       }
 
+      // if that's the end of the root contract's interaction - commit all the uncommitted states to cache.
       if (contract.isRoot()) {
+        // update the uncommitted state of the root contract
+        contract.setUncommittedState(contract.txId(), lastConfirmedTxState.state);
```

Comment:
IDE is suggesting that the lastConfirmedTxState could be null, and considering the line 300 with null checking this suggestion seems to be reasonable.
`if (lastConfirmedTxState !== null) {`